iceoryx_posh/include/iceoryx_posh/internal/popo/building_blocks/chunk_distributor.inl

```diff
@@ -271,6 +266,10 @@ inline void ChunkDistributor<ChunkDistributorDataType>::cleanup() noexcept
                      nullptr,
                      ErrorLevel::FATAL);
     }
+
+    getMembers()->unlock();
```

Comment:
The currently used mutex has the recursive flag set and this call was the only reason. I just wanted to check if it could be removed and it is possible since this code is only executed once the application died and therefore it cannot get the lock again. The method called after `getMembers()->unlock();` will lock the mutex before the data is accessed but the locking in this method was just to detect if the application held the lock when it died.